.github/workflows/main.yml

```diff
@@ -158,83 +158,73 @@ jobs:
-      with:
-        python-version: ${{ matrix.python.version }}
-        architecture: ${{ matrix.python.architecture }}
-    - name: Restore cache
-      uses: actions/cache@v3
-      with:
-        path: |
-          ~\appdata\local\pip\cache
-          vendor\bundle
-          .tox
-        key: deps-${{ runner.os }}-${{ matrix.python.architecture }}-${{ hashFiles('requirements/*.txt') }}-${{ matrix.python.version }}
-        restore-keys: |
-          deps-${{ runner.os }}-${{ matrix.python.architecture }}-${{ hashFiles('requirements/*.txt') }}
-          deps-${{ runner.os }}-${{ matrix.python.architecture }}
-    - name: Use old pandas on win32
-      if: matrix.python.architecture
-      # See https://github.com/pandas-dev/pandas/issues/54979
-      run: |
-        (Get-Content .\requirements\coverage.txt) -replace 'numpy==[0-9.]+', 'numpy==1.26.4' | Out-File .\requirements\coverage.txt
-        (Get-Content .\requirements\coverage.txt) -replace 'pandas==[0-9.]+', 'pandas==2.0.3' | Out-File .\requirements\coverage.txt
-        (Get-Content .\requirements\coverage.txt) -replace 'pyarrow==[0-9.]+', '# omitted' | Out-File .\requirements\coverage.txt
-    - name: Install dependencies
-      run: |
-        pip install --upgrade setuptools pip wheel
-        pip install -r requirements/coverage.txt
-        pip install -r requirements/crosshair.txt
-        pip install hypothesis-python/
-    - name: Run tests
-      # remove this ignore when https://github.com/pschanely/hypothesis-crosshair/issues/40 is fixed
-      run: python -m pytest -W "ignore:hypothesis.internal.observability.TESTCASE_CALLBACKS is deprecated" --numprocesses auto ${{ matrix.whichtests == 'nocover' && 'hypothesis-python/tests/nocover' || 'hypothesis-python/tests/ --ignore=hypothesis-python/tests/nocover/ --ignore=hypothesis-python/tests/quality/ --ignore=hypothesis-python/tests/ghostwriter/ --ignore=hypothesis-python/tests/patching/' }}
-
-  test-osx:
-    runs-on: macos-latest
+  cross:
     strategy:
       matrix:
+        os:
+          - windows-latest
+          - macos-latest
+        python-version:
+          - "3.11"
+          - "3.13"
+        python-architecture:
+          - null
+          - "x86"
         task:
-          - check-py310-cover
-          - check-py310-nocover
-          - check-py310-niche
-          - check-py311-cover
+          - cover
+          - nocover
+          - rest
+          - alt-nocover
+          - alt-rest
+        exclude:
+          - os: macos-latest
+            python-architecture: "x86"
+          - python-version: "3.13"
+            python-architecture: "x86"
+          - python-version: "3.11"
+            task: nocover
+          - python-version: "3.11"
+            task: rest
+          - python-version: "3.13"
+            task: alt-nocover
+          - python-version: "3.13"
+            task: alt-rest
       fail-fast: false
+    runs-on: ${{ matrix.os }}
+    env:
+      PYTHONWARNDEFAULTENCODING: ""
```

Comment:
is this so that `PYTHONWARNDEFAULTENCODING={env:PYTHONWARNDEFAULTENCODING:1}` works correctly? A comment to that effect would be great if so!